/*
 * Licensed to the Apache Software Foundation (ASF) under one
 * or more contributor license agreements.  See the NOTICE file
 * distributed with this work for additional information
 * regarding copyright ownership.  The ASF licenses this file
 * to you under the Apache License, Version 2.0 (the
 * "License"); you may not use this file except in compliance
 * with the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

/*
 * Protocol Buffers describing the Job API, api for communicating with a runner
 * for job submission over GRPC.
 */

syntax = "proto3";

package org.apache.beam.model.expansion.v1;

option go_package = "construction_v1";
option java_package = "org.apache.beam.model.expansion.v1";
option java_outer_classname = "ExpansionApi";

import "beam_runner_api.proto";

message ExpansionRequest {
  org.apache.beam.model.pipeline.v1.Components components = 1;
  org.apache.beam.model.pipeline.v1.PTransform transform = 2;
  string namespace = 3;
}

message ExpansionResponse {
  org.apache.beam.model.pipeline.v1.Components components = 1;
  string transform_id = 2;
  string error = 10;
}

// Job Service for constructing pipelines
service ExpansionService {
  rpc Construct (ExpansionRequest) returns (ExpansionResponse);
}
